changelog/9277.breaking.rst

```diff
@@ -0,0 +1,3 @@
+The ``pytest.Instance`` collector type has been removed.
+Importing ``pytest.Instance`` or ``_pytest.python.Instance`` returns a dummy type and emits a deprecation warning.
```

Comment:
Personally I think it's would be better to just remove this, since trying with sqlalchemy returned no warning whatsoever, just no collection.
If it was removed we would have at least an attribute error to go by.

See also https://github.com/pytest-dev/pytest/issues/9343#issuecomment-980462563